cloud/aws/transitgateway/mode/traffic.pm

```diff
@@ -0,0 +1,310 @@
+        },
+        'unit'      => 'B'
+    },
+    'BytesOut' => {
+        'output'    => 'Bytes Out',
+        'label'     => 'bytes-out',
+        'nlabel'    => {
+            'absolute'   => 'gateway.traffic.out.bytes',
+            'per_second' => 'gateway.traffic.out.bytespersecond'
+        },
+        'unit'      => 'B'
+    },
+    'PacketsIn' => {
+        'output'    => 'Packets Received (In)',
+        'label'     => 'packets-in',
+        'nlabel'    => {
+            'absolute'   => 'gateway.packets.in.count',
+            'per_second' => 'gateway.packets.in.countpersecond'
+        },
+        'unit'      => ''
+    },
+    'PacketsOut' => {
+        'output'    => 'Packets Sent (Out)',
+        'label'     => 'packets-out',
+        'nlabel'    => {
+            'absolute'   => 'gateway.packets.out.count',
+            'per_second' => 'gateway.packets.out.countpersecond'
+        },
+        'unit'      => ''
+    },
+    'PacketDropCountBlackhole' => {
+        'output'    => 'Packets Drop Blackhole',
+        'label'     => 'packets-drop-blackhole',
+        'nlabel'    => {
+            'absolute'   => 'gateway.packets.blackholedropped.count',
+            'per_second' => 'gateway.packets.blackholedropped.countpersecond'
+        },
+        'unit'      => ''
+    },
+    'PacketDropCountNoRoute' => {
+        'output'    => 'Packets Drop No Route',
+        'label'     => 'packets-drop-noroute',
+        'nlabel'    => {
+            'absolute'   => 'gateway.packets.noroutedropped.count',
+            'per_second' => 'gateway.packets.noroutedropped.countpersecond'
+        },
+        'unit'      => ''
+    }
+);
+
+
+sub custom_metric_calc {
+    my ($self, %options) = @_;
+
+    $self->{result_values}->{timeframe} = $options{new_datas}->{$self->{instance} . '_timeframe'};
+    $self->{result_values}->{value} = $options{new_datas}->{$self->{instance} . '_' . $options{extra_options}->{metric}};
+    $self->{result_values}->{value_per_sec} = $self->{result_values}->{value} / $self->{result_values}->{timeframe};
+    $self->{result_values}->{metric} = $options{extra_options}->{metric};
+    return 0;
+}
+
+sub custom_metric_threshold {
+    my ($self, %options) = @_;
+
+    my $exit = $self->{perfdata}->threshold_check(
+        value       => defined($self->{instance_mode}->{option_results}->{per_sec}) ? $self->{result_values}->{value_per_sec} : $self->{result_values}->{value},
+        threshold   => [
+            { label => 'critical-' . $metrics_mapping{$self->{result_values}->{metric}}->{label} , exit_litteral => 'critical' },
+            { label => 'warning-' . $metrics_mapping{$self->{result_values}->{metric}}->{label}, exit_litteral => 'warning' }
+        ]
+    );
+    return $exit;
+}
+
+sub custom_metric_perfdata {
+    my ($self, %options) = @_;
+
+    $self->{output}->perfdata_add(
+        instances => $self->{instance},
+        label       => $metrics_mapping{$self->{result_values}->{metric}}->{label},

```

Comment:
you can remove that line (it's useless with force new perfdata

Reply:
Nice catch, didn't see that one!
Should be better now